### PR TITLE
added config.json path #115

### LINKED ISSUE #115: Error in scanner.py for python-3.8 (JWOC)
[scanner.py](https://github.com/vinitshahdeo/PortScanner/blob/feature/python3.8/src/single/scanner.py) 

* https://github.com/vinitshahdeo/PortScanner/blob/306d91d5dac999c3b1fa531584854d787b2fec1e/src/single/scanner.py#L33

image
The low and high range is not mentioned which is low=1 and high=8888

* This line of code is giving an error.
with open("config.json") as json_file:
https://github.com/vinitshahdeo/PortScanner/blob/306d91d5dac999c3b1fa531584854d787b2fec1e/src/single/scanner.py#L12
image

### PR DESCRIPTION
**Pull request by Vinamrata1086**
# Description

Added the path of ```config.json``` to src/single/scanner.py for python3.8

## Screenshot

* 
 config json path

config.json

Fixes #115

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added all the three reviewers: @vinitshahdeo, @kashish121 & @ishika1727

### CODE REVIEW

**Review by vinitshahdeo — COMMENTED** [review]
@Kashish121 Have a look into this.

File: src/single/scanner.py [hunk 1]
```diff
@@ -42,18 +55,18 @@
...
 # Checking the time again
 t2 = datetime.now()
 
 # Calculates the difference of time, to see how long it took to run the script
-total =  t2 - t1
+total = t2 - t1
 
 # Printing the information to screen
-print ('Scanning Completed in: ', total)
+print('Scanning Completed in: ', total)
```
Comment (Kashish121) [inline src/single/scanner.py:72]: Linting issue.

File: src/single/scanner.py [hunk 2]
```diff
@@ -42,18 +55,18 @@
...
-    print ("Couldn't connect to server")
+    print("Couldn't connect to server")
     sys.exit()
 
 # Checking the time again
 t2 = datetime.now()
 
 # Calculates the difference of time, to see how long it took to run the script
-total =  t2 - t1
+total = t2 - t1
```
Comment (Kashish121) [inline src/single/scanner.py:69]: Linting issue.

File: src/single/scanner.py [hunk 3]
```diff
@@ -42,18 +55,18 @@
     sys.exit()
 
 except socket.gaierror:
-    print ("Hostname could not be resolved. Exiting")
+    print("Hostname could not be resolved. Exiting")
     sys.exit()
 
 except socket.error:
-    print ("Couldn't connect to server")
+    print("Couldn't connect to server")
```
Comment (Kashish121) [inline src/single/scanner.py:62]: Linting issue.

File: src/single/scanner.py [hunk 4]
```diff
@@ -42,18 +55,18 @@
     sys.exit()
 
 except socket.gaierror:
-    print ("Hostname could not be resolved. Exiting")
+    print("Hostname could not be resolved. Exiting")
```
Comment (Kashish121) [inline src/single/scanner.py:58]: Linting issue. 
`print ("...")` instead of `print("...")`

File: src/single/scanner.py [hunk 5]
```diff
@@ -1,36 +1,49 @@
...
+with open(get_absolute_path('/config.json')) as config_file:
+    config = json.load(config_file)
+    loRange = int(config['range']['low'])
+    hiRange = int(config['range']['high'])
+
+
+# scanning the port in range specified in config.json
 try:
-    for port in range(loRange,hiRange):  
+    for port in range(loRange, hiRange):
```
Comment (Kashish121) [inline src/single/scanner.py:46]: Linting issue. It should be `range(loRange,hiRange):` and not `range(loRange, hiRange): `.
Follow the format of code previously used.

File: src/single/scanner.py [hunk 6]
```diff
@@ -1,36 +1,49 @@
...
+    dir = os.path.dirname(os.path.abspath(__file__))
+    new = os.path.dirname(os.path.dirname(dir))
+    print(new)
+    split_path = relative_path.split("/")
+    absolute_path = os.path.join(new, *split_path)
+    return absolute_path
+
 
 # Check what time the scan started
 t1 = datetime.now()
```
Comment (Kashish121) [inline src/single/scanner.py:35]: No need for more than 1 extra new line.

File: src/single/scanner.py [hunk 7]
```diff
@@ -1,36 +1,49 @@
...
 
 # Print a nice banner with information on which host we are about to scan
-print ("-" * 60)
-print ("Please wait, scanning remote host....", remoteServerIP)
-print ("-" * 60)
+print("-" * 60)
+print("Please wait, scanning remote host....", remoteServerIP)
+print("-" * 60)
+
+
```
Comment (Kashish121) [inline src/single/scanner.py:22]: No need for more than 1 extra new line.

File: src/single/scanner.py [hunk 8]
```diff
@@ -1,36 +1,49 @@
...
-    hiRange = int(data['range']['high'])
-
+subprocess.call('cls', shell=True)
 
 # Ask for input
-remoteServer    = input("Enter a remote host to scan: ")
-remoteServerIP  = socket.gethostbyname(remoteServer)
+remoteServer = input("Enter a remote host to scan: ")
+remoteServerIP = socket.gethostbyname(remoteServer)
+
```
Comment (Kashish121) [inline src/single/scanner.py:15]: No need for more than 1 extra new line, Please remove line 15 or 16.

**Comment by Kashish121** [comment]
Also, add a screenshot of the working code. :)

**Review by Kashish121 — CHANGES_REQUESTED** [review]

File: src/single/scanner.py (hunk 7 above)
Comment (Vinamrata1086) [reply src/single/scanner.py:20]: done

**Review by Kashish121 — APPROVED** [review]
@Vinamrata1086 the code looks just fine. 💯 Please add the required screenshots and this PR will be good to go.

**Comment by Vinamrata1086** [comment]
> @Vinamrata1086 the code looks just fine. 💯 Please add the required screenshots and this PR will be good to go.

I have added the required screenshot [above](https://github.com/vinitshahdeo/PortScanner/pull/116#issue-564173564) with the whole code.

**Pull request merged**